core/src/main/java/org/apache/iceberg/rest/responses/DropNamespaceResponse.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.responses;
+
+import java.util.Objects;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * Represents a REST response to drop a namespace.
+ * <p>
+ * The server may choose to return an error if the namespace is not empty.
+ */
+public class DropNamespaceResponse {
+
+  // For Jackson to properly fail when deserializing, this needs to be boxed.
+  // Otherwise, the boolean is parsed according to "loose" javascript JSON rules.
+  private Boolean dropped;
+
+  public DropNamespaceResponse() {
+    // Required for Jackson deserialization
+  }
+
+  private DropNamespaceResponse(Boolean dropped) {
+    this.dropped = dropped;
+    validate();
+  }
+
+  DropNamespaceResponse validate() {
+    // Use `checkArguemt` so it will throw IllegalArgumentException, just like the other responses
+    Preconditions.checkArgument(dropped != null,
+        "Cannot instantiate a response to dropping a namespace with missing field(s): `dropped`");
+    return this;
+  }
+
+  public boolean isDropped() {
+    return dropped;
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("dropped", dropped)
+        .toString();
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static class Builder {
+    private Boolean dropped = false;
+
+    private Builder() {
+    }
+
+    public Builder dropped(Boolean isDropped) {
+      this.dropped = isDropped;
+      return this;
+    }
+
+    public DropNamespaceResponse build() {
+      return new DropNamespaceResponse(dropped);
```

Comment:
This should check that `dropped` is not null.

Reply:
Yeah, that's why I suggested checking whether dropped is null. I think you want to catch when `dropped` wasn't configured.